pkg/proxy/iptables/proxier.go

```diff
@@ -918,17 +894,21 @@ func (proxier *Proxier) syncProxyRules(reason syncReason) {
 	}
 
 	// Figure out the new services we need to activate.
+	proxier.changeMu.Lock()
 	serviceSyncRequired, hcServices, staleServices := updateServiceMap(
 		proxier.serviceMap, &proxier.serviceChanges)
+	proxier.changeMu.Unlock()
```

Comment:
Those two aren't synchronized between each other at all. I mean that it's technically possible that e.g. service is up-to-date and endpoints are lagging by 1 minute, or the other way around.
So trying to lock them together doesn't matter at all.